File: python/pig_latin_test.py
# Rewrite this in Unit Test
import unittest
from pig_latin import translate

class PigLatinTestCase(unittest.TestCase):
    """Tests for pig_latin.py"""

    def test_begin_vowel(self):
        self.assertEqual(translate('apple'), 'appleay')
    
    def test_begin_consonant(self):
        self.assertEqual(translate('banana'), 'ananabay')
    
    def test_begin_two_consonants(self):
        self.assertEqual(translate('cherry'), 'errychay')
    
    def test_two_words(self):
        self.assertEqual(translate('eat pie'), 'eatay iepay')

    def test_begin_three_consonants(self):
        self.assertEqual(translate('three'), 'eethray')
    
    def test_count_sch_as_single_phoneme(self):
        self.assertEqual(translate('school'), 'oolschay')
    
    def test_count_qu_as_single_phoneme(self):
        self.assertEqual(translate('quiet'), 'ietquay')
    
    def test_count_qu_as_consonant_when_preceded_by_consonant(self):
        self.assertEqual(translate('square'), 'aresquay')
    
    def test_translate_many_words(self):
        self.assertEqual(translate('the quick brown fox'), 'ethay ickquay ownbray oxfay')

# write a test asserting that capitalized words are still capitalized
# (but with a different initial capital letter, of course) retain the
# punctuation from the original phrase

    def cap_are_still_cap(self):
        self.assertEqual(translate('SQUARE'), 'ARESQUAY')
    
if __name__ == '__main__':
    unittest.main()

File: js/pigLatin.test.js
// Rewrite this in Unit Test

const pig = require("./pigLatin");

test('translates a word beginning with a vowel', () => {
  expect(pig("apple")).toEqual("appleay");
});

test('translates a word beginning with a consonant', () => {
  expect(pig("banana")).toEqual("ananabay");
});

test('translates a word beginning with two consonants', () => {
  expect(pig("cherry")).toEqual("errychay");
});

test('translates two words', () => {
  expect(pig("eat pie")).toEqual("eatay iepay");
});

test('translates a word beginning with three consonants', () => {
  expect(pig("three")).toEqual("eethray");
});

test('counts sch as a single phoneme', () => {
  expect(pig("school")).toEqual("oolschay");
});

test('counts qu as a single phoneme', () => {
  expect(pig("quiet")).toEqual("ietquay");
});

test('counts qu as a consonant even when its preceded by a consonant', () => {
  expect(pig("square")).toEqual("aresquay");
});

test('translates many words', () => {
  expect(pig("the quick brown fox")).toEqual("ethay ickquay ownbray oxfay");
});

// write a test asserting that capitalized words are still capitalized
// (but with a different initial capital letter, of course) retain the
// punctuation from the original phrase

test('capitals are still capitals', () => {
  expect(pig("SQUARE")).toEqual("ARESQUAY");
});
